libsigner/src/events.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::mpsc::Sender;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use std::net::SocketAddr;
+use std::net::TcpListener;
+use std::net::TcpStream;
+
+use std::io::Read;
+
+use clarity::vm::types::QualifiedContractIdentifier;
+use libstackerdb::StackerDBChunkData;
+
+use serde::{Deserialize, Serialize};
+
+use crate::http::{decode_http_body, decode_http_request};
+use crate::EventError;
+
+/// Event structure for newly-arrived StackerDB data
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct StackerDBChunksEvent {
```

Comment:
Would this code be eligible to be moved to the stacks-core SDK that @stjepangolemac is writing?